plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1146,14 +1146,10 @@ void producer_plugin::plugin_initialize(const boost::program_options::variables_
 
          EOS_ASSERT( vm_total_kb > 0, plugin_config_exception, "Unable to get system virtual memory size (not a Linux?), therefore cannot determine if the system has enough virtual memory for multi-threaded read-only transactions on EOS VM OC");
          // reserve 1 for the app thread, 1 for anything else which might use VM
-         int num_threads_supported = (vm_total_kb - vm_used_kb) / 4200000000 - 2;
-         ilog("vm total in kb: ${total}, vm used in kb: ${used}, number of EOS VM OC threads supported ((vm total - vm used)/4.2 TB - 2): ${supp}", ("total", vm_total_kb) ("used", vm_used_kb) ("supp", num_threads_supported));
-         EOS_ASSERT( num_threads_supported >= my->_ro_thread_pool_size, plugin_config_exception, "--read-only-threads (${th}) greater than number of threads supported for EOS VM OC (${supp}) by the system virtual memory size", ("th", my->_ro_thread_pool_size) ("supp", num_threads_supported) );
-
-         if ( my->_ro_thread_pool_size > my->_ro_max_eos_vm_oc_threads_allowed ) {
-            wlog("--read-only-threads (${th}) greater than maximum number of threads allowed (${allowed}) for EOS Vm OC. Set it to ${allowed}", ("th", my->_ro_thread_pool_size) ("allowed", my->_ro_max_eos_vm_oc_threads_allowed));
-            my->_ro_thread_pool_size = my->_ro_max_eos_vm_oc_threads_allowed;
-         }
+         uint16_t num_threads_supported = (vm_total_kb - vm_used_kb) / 4200000000 - 2;
```

Comment:
Why change from `int` to `uint16_t`? Don't we have a potential overflow here if `(vm_total_kb - vm_used_kb) / 4200000000` < 2?

Reply:
The std::min below requires the same type. The intermediate computational results on the right hand will not be stored on the left hand during calculation. But to be safe in future machines which can have support 65535 threads, I will change to uint32_t. Thanks.

Reply:
thanks.

Reply:
@linh2931 your change does not fix the issue https://godbolt.org/z/aE7cE15xs

Reply:
link works for me. Thy this, you get 16 instead of the correct answer 1 (because `num_threads_supported` == 65535)

```
#include <iostream>

int main() {
   size_t vm_total_kb = 5200000000;
   size_t vm_used_kb = 10;
   uint32_t _ro_max_eos_vm_oc_threads_allowed = 16;
   
   uint32_t num_threads_supported = (vm_total_kb - vm_used_kb) / 4200000000;
   num_threads_supported -= 2;
   auto actual_threads_allowed = std::min(_ro_max_eos_vm_oc_threads_allowed, num_threads_supported);
   std::cout << actual_threads_allowed << '\n';
}
```

Reply:
This assert `EOS_ASSERT( num_threads_supported > 2` will prevent the execution from going further.

Reply:
Yes, that's fine, but it should be `EOS_ASSERT( num_threads_supported >= 2`.

Reply:
Message is still not right: `Available system virtual memory cannot support mininum of 3 minimum threads`.

Maybe something like:

`"With the EOS VM OC configured, there is not enough system virtual memory to support the required minimum of 3 threads`